From ff91e03bf6c1e7ac56d85ceaab692e89b529bd34 Mon Sep 17 00:00:00 2001
Message-Id: <ff91e03bf6c1e7ac56d85ceaab692e89b529bd34.1427265330.git.chang-joon.lee@intel.com>
In-Reply-To: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
References: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 17 Mar 2015 12:56:18 +0530
Subject: [PATCH 09/22] SQUASHME! [VPG]: drm/i915: Cancel media promotion
 timer wq during suspend.

SQUASHME! - This patch should be squashed into the following existing patch:
commit 680ecbaf6db3f2e5832f413198a8dd1f1e77c2fd
Change-Id: I18069d753f6e2a4735f86c0d203bbb971c9f62f7
Author: Deepak S <deepak.s@intel.com>
Date:   Thu Feb 20 21:42:37 2014 +0530

    FOR_UPSTREAM [VPG]: drm/i915: Modifying RC6 Promotion timer for
    Media workloads on VLV.

Issue: GMINL-7105
Change-Id: I1c540da94164dd697f55cb849df16e9d2d145e31
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index ffb6b77..79dc033 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3895,11 +3895,16 @@ static void gen6_disable_rps(struct drm_device *dev)
 
 static void cherryview_disable_rps(struct drm_device *dev)
 {
+	struct drm_i915_private *dev_priv = dev->dev_private;
+
 	/* Disable rc6 */
 	vlv_set_rc6_mode(dev, true);
 
 	/* Disable rps */
 	vlv_set_rps_mode(dev, true);
+
+	/* Cancel pending work-item */
+	cancel_delayed_work_sync(&dev_priv->rps.vlv_media_timeout_work);
 }
 
 static void valleyview_disable_rps(struct drm_device *dev)
-- 
1.7.9.5

